### PR TITLE
added input normalization and tests

### PR DESCRIPTION
**Pull request by zwade**
tests include common flag formats and potential edge cases

### CODE REVIEW

**Comment by zwade** [comment]
addresses #4

File: scoreboard/utils.py [hunk 1]
```diff
@@ -118,6 +119,18 @@ def absolute_url(path):
     """Build an absolute URL.  Not safe for untrusted input."""
     return urlparse.urljoin(flask.request.host_url, path)
 
+def normalize_input(answer):
+    """"Take a string and normalize it to a standard format."""
+
+    """Strip leading and trailing whitespace characters"""
+    match = re.match("^\\s*(.*\\S)\\s*$", answer)
```
Comment (k9delight) [inline scoreboard/utils.py]: re.match("^\\s*(.*\\S)\\s*$", answer)

match has an implicit ^

https://docs.python.org/2/library/re.html

**Comment by Matir** [comment]
CLABot does not like the web interface.

File: scoreboard/tests/unit.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (Matir) [inline scoreboard/tests/unit.py]: Can you rename this to scoreboard/tests/utils_test.py?  I want to keep tests for each python module separate.

File: scoreboard/rest.py [hunk 3]
```diff
@@ -441,11 +442,12 @@ def get(self):
     def post(self):
         data = flask.request.get_json()
         unlocked = data.get('unlocked', False)
+        normal = utils.normalize_input(data['answer'])
```
Comment (Matir) [inline scoreboard/rest.py]: normal is a bad variable name: it's not clear *what* is normal.  Just 'answer' would be fine here (and elsewhere in this file)

File: scoreboard/utils.py (hunk 1 above)
Comment (Matir) [inline scoreboard/utils.py]: For just removing leading and trailing whitespace, why not use .strip()?

I think this can be as simple as:

    return answer.strip().lower()

Comment (zwade) [inline scoreboard/utils.py]: ahh, I thought that strip only removed trailing. I'll use this instead

**Comment by zwade** [comment]
@googlebot confirm

File: scoreboard/utils.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 import hashlib
 import hmac
 import os
+import re
```
Comment (Matir) [inline scoreboard/utils.py]: re import no longer needed

File: scoreboard/utils.py [hunk 5]
```diff
@@ -118,6 +119,11 @@ def absolute_url(path):
     """Build an absolute URL.  Not safe for untrusted input."""
     return urlparse.urljoin(flask.request.host_url, path)
 
+def normalize_input(answer):
+    """"Take a string and normalize it to a standard format."""
+
+    """Strip leading and trailing whitespace characters"""
```
Comment (Matir) [inline scoreboard/utils.py]: Please use a single docstring.

File: main.py [hunk 6]
```diff
@@ -29,5 +29,9 @@
         from scoreboard.tests import data
         models.db.create_all()
         data.create_all()
+    elif 'runtests' in sys.argv:
+        from scoreboard.tests.unit import test_all
```
Comment (Matir) [inline main.py]: This isn't going to work with the new module name.  We should have a central test runner, but I don't know the best way to do that.

File: main.py [hunk 7]
```diff
@@ -29,5 +29,9 @@
         from scoreboard.tests import data
         models.db.create_all()
         data.create_all()
+    elif 'runtests' in sys.argv:
+        from scoreboard.tests.utils_test import test_all
```
Comment (zwade) [inline main.py]: This is just a temporary measure, I agree that we should establish a centralized testing system
